packages/frontend/src/components/staking/components/ValidatorBox.js

```diff
@@ -190,8 +199,20 @@ export default function ValidatorBox({
             {amount &&
                 <div className='right'>
                     {staking && <div><Translate id='staking.validatorBox.staking' /></div>}
+                    {farming && <div><Translate id='staking.validatorBox.farming' /></div>}
                     <div className='amount'>
-                        <Balance amount={amount} showBalanceInUSD={showBalanceInUSD} />
+                        {!token ? (
```

Comment:
Ideally, whatever code is rendering the box component would make a decision to render the correct component with the correct properties rather than make our `ValidatorBox` component have an implicit fall-through based on whether a property is undefined or not.  

For example, in this case _everywhere_ other than the new Modal, `ValidatorBox` receives a balance of NEAR tokens and displays them; _only_ the new modal needs to render the TokenAmount instead, and we know _in the context of the modal_ that we will want to render the `TokenAmount`  component instead of the `Balance` component, because we know what type of data we are working with in that modal context.

We try to avoid optional arguments like this in general so that our components have very specific roles and we avoid default/fallback behaviour based on falsy checks wherever we can.

In this case, rather than make `ValidatorBox` show either NEAR tokens _or_ fungible tokens based on a falsy check of one property, please create a separate component for `ValidatorBoxFarmReward` and rename `ValidatorBox` to e.g. `ValidatorBoxNEARReward` :)

Also, this would seem to be redundant as well; you are already rendering the `TokenAmount` directly in the modal itself, so making the `ValidatorBox` component also render it is redundant.  Maybe this box shouldn't even be rendered in the modal if you are going to render a `TokenAmount` component explicitly in the modal itself.